docs/architecture/adr/ADR040-cluster-stability-node-replacement.md

```diff
@@ -0,0 +1,68 @@
+# ADR040: Ensuring cluster stability while replacing nodes
+
+## Status
+
+Pending
+
+## Context
+
+SREs sometimes need to change various things about a cluster's worker nodes, and therefore replace all of the nodes:
+* AMI (e.g. for an EKS update)
+* instance type
+* anything else in the launch template in some way (e.g. instance role)
+In addition to this, in our Terraform we get the latest AMI ID for our EKS version from AWS - this means that as soon as AWS releases a new AMI for our current EKS version, the next time the cluster's deployer step runs for Terraform it will also replace all the nodes.
```

Comment:
```suggestion

In addition to this, in our Terraform we get the latest AMI ID for our EKS version from AWS - this means that as soon as AWS releases a new AMI for our current EKS version, the next time the cluster's deployer step runs for Terraform it will also replace all the nodes.
```